File: problems/001-anagram-detection/java/Optimized.java
import java.util.*;
public class Optimized implements Solution {
    public Object solve(Map<String, Object> c) {
        String s = (String) c.get("s"), t = (String) c.get("t");
        if (s.length() != t.length()) return false;
        int[] count = new int[128];
        for (char ch : s.toCharArray()) count[ch]++;
        for (char ch : t.toCharArray()) if (--count[ch] < 0) return false;
        return true;
    }
}

File: problems/001-anagram-detection/go/optimized.go
package main

func optimized(c map[string]any) any {
	s, t := c["s"].(string), c["t"].(string)
	if len(s) != len(t) {
		return false
	}
	count := map[rune]int{}
	for _, ch := range s {
		count[ch]++
	}
	for _, ch := range t {
		count[ch]--
		if count[ch] < 0 {
			return false
		}
	}
	return true
}
